### PR TITLE
Choose a reasonable size for bucket_count in Elf.cpp

### PR DESCRIPTION
**Pull request by steven-johnson**
(Sadly, this doesn’t actually fix the problem I was hoping to fix, but
might as well keep it)

### CODE REVIEW

File: src/Elf.cpp [hunk 1]
```diff
@@ -772,19 +773,54 @@ std::vector<char> write_shared_object_internal(Object &obj, Linker *linker, cons
     hash.set_alignment(4);
     hash.set_flag(Section::SHF_ALLOC);
     size_t sym_count = syms.size();
+    std::vector<unsigned long> sym_hashes;
```
Comment (dsharletg) [inline src/Elf.cpp]: The logic I've seen used in LLVM's linker is to simply set bucket_count = sym_count. That would be a lot simpler :)

File: src/Elf.cpp [hunk 2]
```diff
@@ -772,19 +773,54 @@ std::vector<char> write_shared_object_internal(Object &obj, Linker *linker, cons
...
+        bucket_count = unique_hashes / 2;
+    } else {
+        bucket_count = std::max(unique_hashes, (size_t) 1);
+    }
+
+    debug(2) << "sym_count " << sym_count 
+            << " unique_hashes " << unique_hashes 
+            << " bucket_count " << bucket_count << "\n";
+
     // TODO: Fix non-trivial hash tables so they work with dlsym.
```
Comment (dsharletg) [inline src/Elf.cpp]: Remove this TODO

**Comment by steven-johnson** [comment]
Heh -- I misunderstood the comment, I thought it was saying "dlsym doesn't understand this because bucket_count = 1". I didn't even check to see if dlsym works. 

Let's just leave this open-but-unmerged for now; I was basically just trying to see if this might help an odd problem I was trying to debug (it didn't), but figured it was more-right than what we had before. Sounds like that might not be the case.

**Comment by steven-johnson** [comment]
I'm gonna close this since it's unnecessary and possibly misguided. We can resurrect it later if it proves useful.

**Pull request closed without merging**